src/transporters.ts

```diff
@@ -95,6 +98,13 @@ export class DefaultTransporter {
       }
     }
 
+    // If the user configured an `HTTPS_PROXY` environment variable, create
+    // a custom agent to proxy the request.
+    const proxy = process.env.HTTPS_PROXY || process.env.https_proxy;
+    if (proxy) {
+      opts.httpsAgent = new HttpsProxyAgent(proxy);
```

Comment:
In my testing, this only works if you also pass `proxy: false` as an option

Reply:
When I change the code to this, it works:
```javascript
        const proxy = process.env.HTTPS_PROXY || process.env.https_proxy;
        if (proxy) {
            opts.httpsAgent = new HttpsProxyAgent(proxy);
            opts.proxy = false
        }
```

Reply:
Actually, I might be wrong. Retesting

Reply:
OK I _was_ right.

Reply:
Can you share how you're testing this?

Reply:
We have some proxy server running somewhere with access logs. Ive set `HTTPS_PROXY=http://localhost:31280` (Im using SSH port-forwarding to access it) in the environment for my code.

When I use my original hack from #352 , I can connect to Google and I see the access logged on the proxy server. That hack adds to the axios config: 
```typescript
{
        httpsAgent,
        proxy: false
    }
```

When I `npm install 'JustinBeckwith/google-auth-library-nodejs#proxy'` and comment out my hack, the proxy returns an error.

If I then go into `node_modules/google-auth-library-nodejs/build/src/transporters.js` and add `opts.proxy = false` after your initial `opts.httpsAgent = new HttpsProxyAgent(proxy);`, I can once again connect to Google and the proxy logs the access.